src/ngx_http_lua_args.c

```diff
@@ -197,23 +201,27 @@ ngx_http_lua_ngx_req_get_post_args(lua_State *L)
 
     dd("post body length: %d", (int) len);
 
-    buf = ngx_palloc(r->pool, len);
-    if (buf == NULL) {
-        return luaL_error(L, "no memory");
-    }
+    retval = 0;
```

Comment:
It looks wrong to return no values when `len == 0` since we used to return an empty Lua table in this case.